metrics-healthchecks/src/main/java/com/codahale/metrics/health/HealthCheck.java

```diff
@@ -146,11 +185,83 @@ public String toString() {
             if (error != null) {
                 builder.append(", error=").append(error);
             }
+            if (details != null) {
+                Iterator<Map.Entry<String, Object>> it = details.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry<String, Object> e = it.next();
+                    builder.append(e.getKey())
+                        .append("=")
+                        .append(e.getValue().toString());
+                }
+            }
             builder.append('}');
             return builder.toString();
         }
     }
 
+	/**
+     * This a convenient builder for an {@link HealthCheck.Result}. It can be health (with optional message and detail)
+     * or unhealthy (with optional message, error and detail)
+     */
+    public static class ResultBuilder {
+        private boolean healthy;
+        private String message;
+        private Throwable error;
+        private Map<String, Object> details;
+
+        protected ResultBuilder(boolean healthy) {
+            this.details = new LinkedHashMap<String, Object>();
+            this.healthy = healthy;
+        }
+
+        protected ResultBuilder(Throwable error) {
+            this(false);
+            this.error = error;
+            this.message = error.getMessage();
+        }
+
+		/**
+         * Set an optional message
+         *
+         * @param message an informative message
+         * @return this builder with the given {@code message}
+         */
+        public ResultBuilder withMessage(String message) {
+            this.message = message;
+            return this;
+        }
+
+		/**
+         * Set an optional formatted message
+         * <p/>
+         * Message formatting follows the same rules as {@link String#format(String, Object...)}.
+         *
+         * @param message a message format
+         * @param args    the arguments apply to the message format
+         * @return this builder with the given formatted {@code message}
+         * @see String#format(String, Object...)
+         */
+        public ResultBuilder withMessage(String message, Object... args) {
+            return withMessage(String.format(message, args));
+        }
+
+		/**
+         * Add an optional detail
+         *
+         * @param key a key for this detail
+         * @param data an object representing the detail data
+         * @return this builder with the given detail added
+         */
+        public ResultBuilder withDetail(String key, Object data) {
```

Comment:
I use the same name as spring does in its module spring-actuator, I'm not closed to it. What do you prefer: details ? attributes ? properties ? Then I can adapt the method to be add*** instead of with***